data_source.go

```diff
@@ -138,10 +153,11 @@ func (ds *AnySource) PrepareRun() error {
 	// Launch goroutines to drain the data produced by this source
 	ds.processors = make([]*DataStreamProcessor, ds.nchan)
 	ds.runDone.Add(ds.nchan)
-	for chnum, ch := range ds.output {
-		dsp := NewDataStreamProcessor(chnum, ds.broker)
+	for idnum, dataSegChan := range ds.output {
+		dsp := NewDataStreamProcessor(idnum, ds.broker)
+		dsp.Name = ds.chanNames[idnum]
 		dsp.SampleRate = ds.sampleRate
-		ds.processors[chnum] = dsp
+		ds.processors[idnum] = dsp
```

Comment:
ive been calling this processorIndex, I think we should pick one name and use it consistently, im happy to have you make the decision without further bike shedding

Reply:
actually to match `channelNames` maybe we should use `channelIndex`?